### DISCUSSION TITLE
brackets default function NOT work

### DISCUSSION THREAD

**Discussion by uptownboy7** [brackets-cont/brackets#106] [Q&A]
just see my image.
https://www.gloomy-store.com/react/ww2.png

I Set the .js files to JSX highlighter.

click [.js] => click [.jsx] => click [set default .js file]

and i reset the brackets.

and....what?!
its is **still JS highlighter**, not JSX highlighter.
default not changed.

how can i do for this sir
![ww2](https://user-images.githubusercontent.com/57980360/147842377-9ce6623b-df39-4acf-ab6d-93cc90eedcf3.png)
?

**Comment by abose** [comment]
Can you post `help> health report` here?

**Reply by uptownboy7** [reply]
okay. i can post health report by same post of this. thanx

**Reply by uptownboy7** [reply]
[
    {
        "snapshotTime": 1641426634744,
        "os": "win",
        "userAgent": "Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Brackets/1.14.2.0 Chrome/49.0.2623.110 Safari/537.36",
        "osLanguage": "ko-KR",
        "bracketsLanguage": "ko-KR",
        "bracketsVersion": "2.0.1-17920",
        "fileStats": {
            "openedFileExt": {
                "js": 6,
                "php": 1,
                "html": 2
            },
            "workingSetFileExt": {
                "js": 3,
                "php": 1,
                "html": 1
            },
            "openedFileEncoding": {
                "UTF-8": 9
            }
        },
        "ProjectDetails": {
            "prj-1275470785": {
                "numFiles": 29,
                "cacheSize": 529090
            }
        },
        "projectLoadTimes": ":71",
        "fileOpenTimes": ":160:59:611:100(141)182[41]:94",
        "prefNodeSearchDisabled": false,
        "prefInstantSearchDisabled": false,
        "installedExtensions": [
            {
                "name": "blueprint.outliner",
                "version": "0.9.3"
            },
            {
                "name": "brackets-reverse-emmet",
                "version": "0.0.1"
            },
            {
                "name": "dustindowell.neon-minimap",
                "version": "1.0.1"
            },
            {
                "name": "brackets-php-beautify",
                "version": "0.1.1"
            },
            {
                "name": "quickdocsphp",
                "version": "1.4.7"
            },
            {
                "name": "brackets-emmet",
                "version": "1.2.2"
            },
            {
                "name": "brackets-beautify",
                "version": "2.12.0"
            },
            {
                "name": "mackenza.php-smarthints",
                "version": "1.2.2"
            },
            {
                "name": "brackets-indent-guides",
                "version": "1.3.11"
            },
            {
                "name": "custom-work-for-brackets",
                "version": "5.2.0"
            }
        ],
        "bracketsTheme": "dark-theme",
        "uuid": "67dd-ae35-abf7-5481-0c40",
        "olderuuid": "67dd-ae35-abf7-5481-0c40"
    },
    [
        {
            "eventName": "usagefileOpenHTML",
            "eventCategory": "usage",
            "eventSubCategory": "fileOpen",
            "eventType": "html",
            "eventSubType": ""
        },
        {
            "eventName": "usagefileOpenCSS",
            "eventCategory": "usage",
            "eventSubCategory": "fileOpen",
            "eventType": "css",
            "eventSubType": ""
        },
        {
            "eventName": "livePreviewOpen",
            "eventCategory": "usage",
            "eventSubCategory": "livePreview",
            "eventType": "open",
            "eventSubType": ""
        },
        {
            "eventName": "usagefileSaveCSS",
            "eventCategory": "usage",
            "eventSubCategory": "fileSave",
            "eventType": "css",
            "eventSubType": ""
        },
        {
            "eventName": "usagefileOpenJavaScript",
            "eventCategory": "usage",
            "eventSubCategory": "fileOpen",
            "eventType": "javascript",
            "eventSubType": ""
        },
        {
            "eventName": "usagefileSaveJavaScript",
            "eventCategory": "usage",
            "eventSubCategory": "fileSave",
            "eventType": "javascript",
            "eventSubType": ""
        },
        {
            "eventName": "refactoring.extractToVariable",
            "eventCategory": "usage",
            "eventSubCategory": "jsRefactor",
            "eventType": "extractToVariable",
            "eventSubType": ""
        },
        {
            "eventName": "usagefileCloseHTMLSize_0_10KB",
            "eventCategory": "usage",
            "eventSubCategory": "fileClose",
            "eventType": "html",
            "eventSubType": "Size_0_10KB"
        },
        {
            "eventName": "usagefileSaveHTML",
            "eventCategory": "usage",
            "eventSubCategory": "fileSave",
            "eventType": "html",
            "eventSubType": ""
        },
        {
            "eventName": "usagefileCloseJavaScriptSize_10_50KB",
            "eventCategory": "usage",
            "eventSubCategory": "fileClose",
            "eventType": "javascript",
            "eventSubType": "Size_10_50KB"
        },
        {
            "eventName": "multiCursor",
            "eventCategory": "usage",
            "eventSubCategory": "multiCursor",
            "eventType": "use",
            "eventSubType": ""
        },
        {
            "eventName": "usagefileOpennewfile",
            "eventCategory": "usage",
            "eventSubCategory": "fileOpen",
            "eventType": "newfile",
            "eventSubType": ""
        },
        {
            "eventName": "usagefileOpenPHP",
            "eventCategory": "usage",
            "eventSubCategory": "fileOpen",
            "eventType": "php",
            "eventSubType": ""
        },
        {
            "eventName": "usagefileSaveText",
            "eventCategory": "usage",
            "eventSubCategory": "fileSave",
            "eventType": "text",
            "eventSubType": ""
        },
        {
            "eventName": "usagefileSavePHP",
            "eventCategory": "usage",
            "eventSubCategory": "fileSave",
            "eventType": "php",
            "eventSubType": ""
        },
        {
            "eventName": "usagefileClosePHPSize_0_10KB",
            "eventCategory": "usage",
            "eventSubCategory": "fileClose",
            "eventType": "php",
            "eventSubType": "Size_0_10KB"
        },
        {
            "eventName": "usagefileOpenSQL",
            "eventCategory": "usage",
            "eventSubCategory": "fileOpen",
            "eventType": "sql",
            "eventSubType": ""
        },
        {
            "eventName": "usagelanguageChangeJSXexistingFile",
            "eventCategory": "usage",
            "eventSubCategory": "languageChange",
            "eventType": "jsx",
            "eventSubType": "existingFile"
        }
    ]
]

**Reply by uptownboy7** [reply]
If i open new js file, brackets shows js highlighter but not jsx highlighter even i set jsx highlighter to default for js file.

**Comment by abose** [comment]
Duplicate of https://github.com/brackets-cont/brackets/discussions/116
